### PR TITLE
MM-14194: fix subpath csp directive until server restart

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
The SiteURL organically doesn't take effect until server restart, but in v5.8, the required CSP directive would change immediately. If changing from one subpath to another, the webapp would effectively be bricked until a server restart.

Avoid this by determining the CSP directive when the static handler is created.

I'm not super happy about the inability to unit test this in v5.9, but the foundation for "doing this properly" is almost present in master, so I'm content to leave the changes as-is and circle back once my final config store changes are merged.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-14194

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: web/handlers.go [hunk 1]
```diff
@@ -26,13 +26,22 @@ func (w *Web) NewHandler(h func(*Context, http.ResponseWriter, *http.Request)) h
 }
 
 func (w *Web) NewStaticHandler(h func(*Context, http.ResponseWriter, *http.Request)) http.Handler {
+	// Determine the CSP SHA directive needed for subpath support, if any. This value is fixed
+	// on server start and intentionally requires a restart to take effect.
+	app := app.New(
```
Comment (crspeller) [inline web/handlers.go]: Can we pass in the config this needs instead?

Comment (lieut-data) [reply web/handlers.go]: Looks like my original approach was convoluted: I can just get it off of `Web` instead.

**Comment by lieut-data** [comment]
> Also curious what prevents us from making the SiteURL able to be changed without a restart?

This is a good question. I had been under the assumption it was a common limitation, but at least from my perspective, it's not the `SiteURL` per se, but where and when we apply the subpath. Today, we only do that on server startup (if we have access to the `client/` directory). Conceivably, we could listen for config changes and rewrite the SiteURL in real-time. This wouldn't apply to a CDN deploy, but then that's a whole different problem in general.

Thoughts?

**Comment by crspeller** [comment]
@lieut-data Seems like that might be a better fix, as long as the implementation doesn't end up being too complicated.

**Comment by lieut-data** [comment]
@crspeller, I looked into this a bit. It's easy enough to rewrite the subpath assets when the config changes. The more difficult challenge is that we [bake the subpath](https://github.com/mattermost/mattermost-server/blob/master/app/server_app_adapters.go#L98) into the router on initialization, and I'm not sure how to reliably re-initialize this part of the server without, well restarting it in the first place.

Thoughts?

**Pull request merged**